hazelcast/include/hazelcast/client/map/MixedMapProxy.h

```diff
@@ -0,0 +1,1136 @@
+                *                     contain the value.
+                */
+                template <typename K>
+                std::string addEntryListener(MixedEntryListener &listener, const K &key, bool includeValue) {
+                    serialization::pimpl::Data keyData = toData(key);
+                    client::impl::MixedEntryEventHandler<protocol::codec::MapAddEntryListenerCodec::AbstractEventHandler> *entryEventHandler =
+                            new client::impl::MixedEntryEventHandler<protocol::codec::MapAddEntryListenerCodec::AbstractEventHandler>(
+                                    getName(), context->getClusterService(), context->getSerializationService(),
+                                    listener,
+                                    includeValue);
+                    return proxy::IMapImpl::addEntryListener(entryEventHandler, keyData, includeValue);
+                }
+
+                /**
+                * Returns the <tt>EntryView</tt> for the specified key.
+                *
+                *
+                * @param key key of the entry
+                * @return <tt>EntryView</tt> of the specified key
+                * @see EntryView
+                */
+                EntryView <TypedData, TypedData> getEntryView(const TypedData &key);
+
+                /**
+                * Evicts the specified key from this map. If
+                * a <tt>MapStore</tt> defined for this map, then the entry is not
+                * deleted from the underlying <tt>MapStore</tt>, evict only removes
+                * the entry from the memory.
+                *
+                *
+                * @param key key to evict
+                * @return <tt>true</tt> if the key is evicted, <tt>false</tt> otherwise.
+                */
+                template <typename K>
+                bool evict(const K &key) {
+                    serialization::pimpl::Data keyData = toData(key);
+
+                    return evictInternal(keyData);
+                }
+
+                /**
+                * Evicts all keys from this map except locked ones.
+                * <p/>
+                * If a <tt>MapStore</tt> is defined for this map, deleteAll is <strong>not</strong> called by this method.
+                * If you do want to deleteAll to be called use the #clear() method.
+                * <p/>
+                * The EVICT_ALL event is fired for any registered listeners.
+                * See EntryListener#mapEvicted(MapEvent)}.
+                *
+                * @see #clear()
+                */
+                void evictAll();
+
+                /**
+                * Returns the entries for the given keys.
+                *
+                * @param keys keys to get
+                * @return map of entries
+                */
+                template <typename K>
+                std::vector<std::pair<TypedData, TypedData> > getAll(const std::set<K> &keys) {
```

Comment:
Isn't this gonna cause problems ? 
We have a class in vector that does not copy properly -> When copied old data is useless.

Reply:
yes, that may be a problem in the future, since it is a user api. I changed the copy constructor of TypedData to perform deep copy of bytes.